crates/oxc_linter/src/rules/import/extensions.rs

```diff
@@ -179,220 +326,407 @@ declare_oxc_lint!(
+/// - Package subpaths: `lodash/fp`, `@babel/core/lib/parser`
+///
+/// Returns `false` for:
+/// - Relative imports: `./foo`, `../bar`
+/// - Absolute paths: `/usr/local/lib`
+/// - Path aliases: `@/`, `~/`, `#/`
+fn is_package_import(module_name: &str) -> bool {
+    // Relative imports: ./foo, ../bar, or directory imports (., ..)
+    if module_name.starts_with('.') {
+        return false;
+    }
+
+    // Absolute paths: /foo
+    if module_name.starts_with('/') {
+        return false;
+    }
+
+    // Handle @ prefix: distinguish path aliases from scoped packages
+    // - @/foo (path alias) → rest = "/foo" → starts with '/'
+    // - @x/pkg (scoped package) → rest = "x/pkg" → doesn't start with '/'
+    // - @babel/core (scoped package) → rest = "babel/core" → doesn't start with '/'
+    if let Some(rest) = module_name.strip_prefix('@') {
+        if rest.starts_with('/') {
+            return false; // Path alias: @/
         }
+        // Scoped packages must have scope/package format
+        // This includes single-letter scopes like @x/pkg
+        return rest.contains('/');
     }
 
-    fn process_require_record(
-        &self,
-        call_expr: &CallExpression<'_>,
-        ctx: &LintContext,
-        require_extension: Option<&FileExtensionConfig>,
-    ) {
-        let config = &self.0;
-        for argument in &call_expr.arguments {
-            if let Argument::StringLiteral(s) = argument {
-                let file_extension = get_file_extension_from_module_name(&s.value.to_compact_str());
-                let span = call_expr.span;
-
-                if let Some(file_extension) = file_extension {
-                    let ext_str = file_extension.as_str();
-                    let should_flag = match require_extension {
-                        Some(FileExtensionConfig::Always) => {
-                            config.is_never(ext_str) || !config.is_always(ext_str)
-                        }
-                        Some(FileExtensionConfig::Never) => !config.is_always(ext_str),
-                        _ => config.is_never(ext_str),
-                    };
-
-                    if should_flag {
-                        ctx.diagnostic(extension_should_not_be_included_in_diagnostic(
-                            span,
-                            &file_extension,
-                            true,
-                        ));
-                    }
-                } else if matches!(
-                    require_extension,
-                    Some(FileExtensionConfig::Always | FileExtensionConfig::IgnorePackages)
-                ) {
-                    ctx.diagnostic(extension_missing_diagnostic(span, true));
-                }
-            }
+    // Other single-char path aliases: ~/, #/
+    if module_name.len() >= 2 {
+        let bytes = module_name.as_bytes();
+        if bytes[1] == b'/' && bytes[0] != b'.' && bytes[0] != b'@' {
+            return false; // Path alias like ~/ or #/
         }
     }
```

Comment:
Note for follow-up: what's the faster way to peek the first two bytes of a str? Haven't checked the assembly, but I wonder if this incurs three bounds checks instead of just one.